File: 0x01-lockboxes/main_.js
import canUnlockAll from "./0-lockboxes.js";

const boxes = [[1], [2], [3], [4], []];
console.log(canUnlockAll(boxes));

const boxes1 = [[1, 4, 6], [2], [0, 4, 1], [5, 6, 2], [3], [4, 1], [6]];
console.log(canUnlockAll(boxes1));

const boxes2 = [[1, 4], [2], [0, 4, 1], [3], [], [4, 1], [5, 6]]
console.log(canUnlockAll(boxes2));

console.log(canUnlockAll([[1], "", [], [1]]));

File: 0x01-lockboxes/main_.py
#!/usr/bin/python3

canUnlockAll = __import__('0-lockboxe').canUnlockAll

boxes = [[1], [2], [3], [4], []]
print(canUnlockAll(boxes))

boxes = [[4], [2, 7], [0, 4, 1], [5, 6, 2], [3], [4, 1], [1, 4, 6]]
print(canUnlockAll(boxes))

boxes = [[1, 4], [2, 4], [0, 4, 1], [3, 1], [], [4, 1], [5, 6]]
print(canUnlockAll(boxes))
print(canUnlockAll([[1], "", [], [1]]))
